server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -193,7 +193,7 @@ public IndexService(
                 // we delay the actual creation of the sort order for this index because the mapping has not been merged yet.
                 // The sort order is validated right after the merge of the mapping later in the process.
                 this.indexSortSupplier = () -> indexSettings.getIndexSortConfig().buildIndexSort(
-                    mapperService::fieldType,
+                    mapperService::concreteFieldType,
```

Comment:
We can't detect runtime fields using a SearchLookup implementation any more, so instead I've added this method.